app/src/main/java/nodomain/freeyourgadget/gadgetbridge/service/devices/liveview/LiveviewIoThread.java

```diff
@@ -0,0 +1,220 @@
+package nodomain.freeyourgadget.gadgetbridge.service.devices.liveview;
+
+import android.bluetooth.BluetoothAdapter;
+import android.bluetooth.BluetoothDevice;
+import android.bluetooth.BluetoothSocket;
+import android.content.Context;
+import android.os.ParcelUuid;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.nio.ByteBuffer;
+import java.nio.ByteOrder;
+import java.util.UUID;
+
+import nodomain.freeyourgadget.gadgetbridge.deviceevents.GBDeviceEvent;
+import nodomain.freeyourgadget.gadgetbridge.impl.GBDevice;
+import nodomain.freeyourgadget.gadgetbridge.service.serial.GBDeviceIoThread;
+import nodomain.freeyourgadget.gadgetbridge.service.serial.GBDeviceProtocol;
+import nodomain.freeyourgadget.gadgetbridge.util.GB;
+
+public class LiveviewIoThread extends GBDeviceIoThread {
+    private static final Logger LOG = LoggerFactory.getLogger(LiveviewIoThread.class);
+
+    private static final UUID SERIAL = UUID.fromString("00001101-0000-1000-8000-00805F9B34FB");
+
+    private final LiveviewProtocol mLiveviewProtocol;
+    private final LiveviewSupport mLiveviewSupport;
+
+
+    private BluetoothAdapter mBtAdapter = null;
+    private BluetoothSocket mBtSocket = null;
+    private InputStream mInStream = null;
+    private OutputStream mOutStream = null;
+    private boolean mQuit = false;
+
+    @Override
+    public void quit() {
+        mQuit = true;
+        if (mBtSocket != null) {
+            try {
+                mBtSocket.close();
+            } catch (IOException e) {
+                e.printStackTrace();
```

Comment:
I'd love to see the number of e.printStackTrace() decrease rather than increase :-]